File: Composicion/Exercise1/Java-Eclipse/src/composicion/Habitacion.java
package composicion;
import java.util.ArrayList;

public class Habitacion {
	private String nombre;
    private double tamano;
    
    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }
    
    public String getNombre() {
        return nombre;
    }
    
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    
    public double getTamano() {
        return tamano;
    }
    
    public void setTamano(double tamano) {
        this.tamano = tamano;
    }
    
    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}
class Casa {
    private String direccion;
    private java.util.ArrayList<Habitacion> habitaciones;
    
    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new java.util.ArrayList<>();
    }
    
    public String getDireccion() {
        return direccion;
    }
    
    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }
    
    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }
    
    public void mostrarCasa() {
        System.out.println("\nCasa en: " + direccion);
        System.out.println("Habitaciones:");
        for (Habitacion habitacion : habitaciones) {
            habitacion.mostrarInfo();
        }
    }
}

File: Composicion/Exercise1/Python/Habitacion.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.__nombre = nombre
        self.__tamano = tamano
    
    @property
    def nombre(self):
        return self.__nombre
    
    @property
    def tamano(self):
        return self.__tamano
    
    @nombre.setter
    def nombre(self, nombre):
        self.__nombre = nombre
    
    @tamano.setter
    def tamano(self, tamano):
        self.__tamano = tamano
    
    def mostrar_info(self):
        print(f"Habitación: {self.__nombre}, Tamaño: {self.__tamano} m²")

class Casa:
    def __init__(self, direccion):
        self.__direccion = direccion
        self.__habitaciones = []
    
    
    @property
    def direccion(self):
        return self.__direccion
    
    @property
    def habitaciones(self):
        return self.__habitaciones.copy()  
    
    # Setters
    @direccion.setter
    def direccion(self, direccion):
        self.__direccion = direccion
    
    def agregar_habitacion(self, habitacion):
        self.__habitaciones.append(habitacion)
    
    def mostrar_casa(self):
        print(f"\nCasa en: {self.__direccion}")
        print("Habitaciones:")
        for p in self.__habitaciones:
            p.mostrar_info()

sa = Habitacion("Sala", 20.5)
co = Habitacion("Cocina", 15.0)
ba = Habitacion("Baño", 8.5)
do = Habitacion("Dormitorio principal", 18.0)

mi_casa = Casa("Calle Genaro Sanjines 123")

mi_casa.agregar_habitacion(sa)
mi_casa.agregar_habitacion(co)
mi_casa.agregar_habitacion(ba)
mi_casa.agregar_habitacion(do)

mi_casa.mostrar_casa()